packages/x-date-pickers/src/DateField/useDateField.ts

```diff
@@ -111,5 +112,5 @@ export const useDateField = <TDate, TChildProps extends {}>({
     fieldValueManager: dateFieldValueManager,
     validator: validateDate,
     supportedDateSections: ['year', 'month', 'day'],
-  });
+  }) as any;
```

Comment:
OK I see two errors in the typing:

1. We should remove the `inputRef` from `internalProps`, it is now passed at the root of the params (you can see that we are passing it twice)

2. `forwardedProps` contains no known props so TypeScript can't know if what we are passing to it matches the expected type. We can do `forwardedProps: other as unknown as TChildProps,` to force it to work without an `as any` on the whole response.

```diff
@@ -89,7 +90,7 @@ export const useDateField = <TDate, TChildProps extends {}>({
 
   return useField({
     inputRef,
-    forwardedProps: other,
+    forwardedProps: other as unknown as TChildProps,
     internalProps: {
       value,
       defaultValue,
@@ -106,11 +107,10 @@ export const useDateField = <TDate, TChildProps extends {}>({
       disablePast,
       selectedSections,
       onSelectedSectionsChange,
-      inputRef,
     },
     valueManager: datePickerValueManager,
     fieldValueManager: dateFieldValueManager,
     validator: validateDate,
     supportedDateSections: ['year', 'month', 'day'],
-  }) as any;
+  });
 };
```

The logic is probably the same for the other fields

Reply:
To avoid the casting on `forwardedProps`, I don't have any idea.
Except ugly ones like adding a fake prop to make sure that it has one prop in common.
This "could be instantiated with a different subtype of constraint" is a pain

Reply:
https://askcodes.net/coding/how-to-fix-ts2322---could-be-instantiated-with-a-different-subtype-of-constraint--object---

Nice blog post but don't seem match our setup :cry: